activerecord/lib/active_record/relation/batches.rb

```diff
@@ -252,28 +259,32 @@ def in_batches(of: 1000, start: nil, finish: nil, load: false, error_on_ignore:
         end
 
         batch_relation = relation.where(
-          bind_attribute(primary_key, primary_key_offset) { |attr, bind| attr.gt(bind) }
+          bind_attribute(primary_key, primary_key_offset) { |attr, bind| order == :desc ? attr.lt(bind) : attr.gt(bind) }
         )
       end
     end
 
     private
-      def apply_limits(relation, start, finish)
-        relation = apply_start_limit(relation, start) if start
-        relation = apply_finish_limit(relation, finish) if finish
+      def apply_limits(relation, start, finish, order)
+        relation = apply_start_limit(relation, start, order) if start
+        relation = apply_finish_limit(relation, finish, order) if finish
         relation
       end
 
-      def apply_start_limit(relation, start)
-        relation.where(bind_attribute(primary_key, start) { |attr, bind| attr.gteq(bind) })
+      def apply_start_limit(relation, start, order)
+        relation.where(bind_attribute(primary_key, start) { |attr, bind| order == :desc ? attr.lteq(bind) : attr.gteq(bind) })
       end
 
-      def apply_finish_limit(relation, finish)
-        relation.where(bind_attribute(primary_key, finish) { |attr, bind| attr.lteq(bind) })
+      def apply_finish_limit(relation, finish, order)
+        relation.where(bind_attribute(primary_key, finish) { |attr, bind| order == :desc ? attr.gteq(bind) : attr.lteq(bind) })
       end
 
-      def batch_order
-        arel_attribute(primary_key).asc
+      def batch_order(order)
+        if order == :desc
+          arel_attribute(primary_key).desc
+        else
+          arel_attribute(primary_key).asc
+        end
```

Comment:
```suggestion
        arel_attribute(primary_key).public_send(order)
```

Since we've already validated that order is asc/desc only.